src/main/java/smp/SMPData.java

```diff
@@ -0,0 +1,53 @@
+package smp;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+public class SMPData {
+
+    private int[][] preferences_one;
+    private int[][] preferences_two;
+
+    private SMPData(int[][] pref1, int[][] pref2) {
+        preferences_one = pref1;
+        preferences_two = pref2;
+    }
+
+    public int[][] getPreferencesOne() {
+        return preferences_one;
+    }
+
+    public int[][] getPreferencesTwo() {
+        return preferences_two;
+    }
+
+    public int getSize() {
+        return preferences_one.length;
+    }
+
+    public static SMPData loadFromFile(String filePath) {
```

Comment:
This class does the parsing of the test case, please try to use it so we can get rid of duplicated code for loading input data.